📖✨：revamp MDG description page (#1)
OpenINF/Gherkin-Markdown: MARKDOWN_WITH_GHERKIN.md

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

<section id="introduction">

### Introduction

<br />

The specification for the OpenINF Markdown dialect is based on several of its

> more commonly-known predecessors, such as the [GitHub Flavored Markdown (GFM)

Author: ```suggestion
more well-known predecessors, such as the [GitHub Flavored Markdown (GFM)
```

> more well-known predecessors, such as the [GitHub Flavored Markdown (GFM)
> specification][], which is, in turn, based on the [CommonMark specification][].

Author: ```suggestion
more well-known predecessors, such as the [GitHub Flavored Markdown (GFM)
formal specification][], which is, in turn, based on the [CommonMark
Spec][].

```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

<section id="introduction">

### Introduction

<br />

> 

Author: ```suggestion
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the

> OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).

Author: ```suggestion
OpenINF Community Markdown Style Guide standards (minimal stylistic processing).
```

> OpenINF Community Markdown Style Guide standards (minimal stylistic processing).

Author: ```suggestion
OpenINF Community Markdown Style Guide standards (minimal stylistic processing)
with additional attribution specified in the Acknowledgements section.
```

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

<section id="intro">

### Introduction

<br />

The specification for the OpenINF Markdown dialect is based on several of its
more well-known predecessors, such as the [GitHub Flavored Markdown Spec][],
which is, in turn, based on the [CommonMark Spec][].

A <dfn id="spec">Specification</dfn> (<abbr>spec</abbr>) is a document that
outlines, in detail, how a technology or API is intended to function and how
it is accessed.

The OpenINF Markdown spec includes [many additions][] compared to its
predecessors.

<br /><br />

<section id="official-specs">

#### Official Specifications&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#official-specs" id="ref-for-official-specs">⚓</a>

<br />

GFM and GLFM each have an official specification, which includes both:

1. The CommonMark standard.
1. Generic extensions to the CommonMark standard.

For example, GFM adds the [`strikethrough` extension][], and GLFM adds the
[`color chips` extension][]. In their respective official specifications,
these extensions are not dependent upon any specific implementation or

> environment; therefore, they may be implemented in any rendering engine.

Author: ```suggestion
environment; therefore, they may be implemented in foreign engines.
```

implemented in any third-party Markdown rendering engine.

This document describes OpenINF Markdown with Gherkin (MDG), which some may
expect to detail how Gherkin support is added via generic extension(s) to the
CommonMark standard (as is typically the case). However, with MDG, inserting
Gherkin scenarios directly into Markdown using traditional Markdown syntax is
feasible. As a result, any GFM-compliant library can render MDG documents.

[`strikethrough` extension]: https://github.github.com/gfm/#strikethrough-extension-
[`color chips` extension]: https://docs.gitlab.com/ee/user/markdown.html#colors

> </section>

Author: ```suggestion
</section> <!-- END #official-specifications -->
```

<section id="termdef-mdg">
<!-- markdownlint-disable-next-line line-length -->
<strong><em><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></em></strong>
is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.

> </section>

Author: ```suggestion

</section> <!-- END #termdef-mdg -->
```

language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as

> a clue for editors, MDG files must have the `.feature.md` extension. With MDG,
> inserting Gherkin scenarios directly into Markdown using traditional Markdown
> syntax is feasible. Moreover, any GFM-compliant library can render MDG
> documents.

Author: ```suggestion
a clue for editors, MDG files must have the `.feature.md` extension.
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide standards (minimal stylistic processing)
with additional attribution specified in the Acknowledgements section footer.
-->

<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

> _1<sup>st</sup> edition (August 2023)_

Author: ```suggestion
_1<sup>st</sup> edition (Spring 2024)_
```

---

<br /><br />

<section id="intro">

### Introduction

<br />

The specification for the OpenINF Markdown dialect is based on several of its
more well-known predecessors, such as the [GitHub Flavored Markdown Spec][],
which is, in turn, based on the [CommonMark Spec][].

A <dfn id="spec">Specification</dfn> (<abbr>spec</abbr>) is a document that
outlines, in detail, how a technology or API is intended to function and how

> it is accessed.

Author: ```suggestion
it is accessed or used.
```

it is accessed.

The OpenINF Markdown spec includes [many additions][] compared to its
predecessors.

<br /><br />

<section id="official-specs">

#### Official Specifications&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#official-specs" id="ref-for-official-specs">⚓</a>

<br />

GFM and GLFM each have an official specification, which includes both:

> 1. The CommonMark standard.
> 1. Generic extensions to the CommonMark standard.

Author: ```suggestion
1. The CommonMark standard
1. Generic extensions to the CommonMark standard
```

> For example, GFM adds the [`strikethrough` extension][], and GLFM adds the
> [`color chips` extension][]. In their respective official specifications,
> these extensions are not dependent upon any specific implementation or
> environment; they can be implemented in any third-party Markdown rendering
> engine.

Author: ```suggestion
For example, GFM adds the [`strikethrough` extension][], and GLFM adds the
[`color chips` extension][]. In their respective official specifications,
these extensions are not dependent upon any specific implementation or
environment; therefore, they may be implemented in any third-party Markdown
rendering engine.
```

> [`color chips` extension][]. These extensions in the official specifications are
> not dependent upon any specific implementation or environment. They can be
> implemented in any third-party Markdown rendering engine.

Author: ```suggestion
[`color chips` extension][]. In their respective official specifications,
these extensions are not dependent upon any specific implementation or
environment. They can be implemented in any third-party Markdown rendering
engine.
```

> environment; therefore, they may be implemented in any third-party Markdown
> rendering engine.

Author: ```suggestion
environment; therefore, they may be implemented in any rendering engine.
```

This document describes OpenINF Markdown with Gherkin (MDG), which some may
expect to detail how Gherkin support is added via generic extension(s) to the
CommonMark standard (as is typically the case). However, with MDG, inserting

> Gherkin scenarios directly into Markdown using traditional Markdown syntax is

Author: ```suggestion
Gherkin scenarios directly into Markdown, using traditional Markdown syntax, is
```

feasible. As a result, any GFM-compliant library can render MDG documents.

[`strikethrough` extension]: https://github.github.com/gfm/#strikethrough-extension-
[`color chips` extension]: https://docs.gitlab.com/ee/user/markdown.html#colors

</section> <!-- END #official-specs -->

<section id="termdef-mdg">
<!-- markdownlint-disable-next-line line-length -->

> <p><em><strong><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></strong></em>

Author: ```suggestion
<p><em><strong><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown&nbsp;with&nbsp;Gherkin&nbsp;(<abbr>MDG</abbr>)&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></strong></em>
```

is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.

</section> <!-- END #termdef-mdg -->

It is widely used for writing documentation, specifications, and test cases.
Markdown is a lightweight markup language that utilizes syntax for plain text
formatting and is often utilized for formatting readme files and writing
messages in online discussion forums. Conversely, Gherkin is a plain text
language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as
a clue for editors, MDG files must have the `.feature.md` extension.

> 

Author: ```suggestion

</section> <!-- END #termdef-mdg -->

</section> <!-- END #introduction -->

```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

<section id="intro">

### Introduction

<br />

The specification for the OpenINF Markdown dialect is based on several of its
more well-known predecessors, such as the [GitHub Flavored Markdown Spec][],
which is, in turn, based on the [CommonMark Spec][].

A <dfn id="spec">Specification</dfn> (<abbr>spec</abbr>) is a document that
outlines, in detail, how a technology or API is intended to function and how
it is accessed.

> 
> The OpenINF Markdown specification includes [many additions][] compared to the
> GFM specification.

Author: ```suggestion
The OpenINF Markdown spec includes [many additions][] compared to its
predecessors.
```

<br /><br />

<section id="official-specs">

#### Official Specifications&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#official-specs" id="ref-for-official-specs">⚓</a>

<br />

GFM and GLFM each have an official specification, which includes both:

1. The CommonMark standard.
1. Generic extensions to the CommonMark standard.

For example, GFM adds the [`strikethrough` extension][], and GLFM adds the
[`color chips` extension][]. In their respective official specifications,
these extensions are not dependent upon any specific implementation or

> environment. They can be implemented in any third-party Markdown rendering

Author: ```suggestion
environment; they can be implemented in any third-party Markdown rendering
```

<!-- markdownlint-disable-next-line line-length -->
<strong><em><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></em></strong>
is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.
</section>

It is widely used for writing documentation, specifications, and test cases.
Markdown is a lightweight markup language that utilizes syntax for plain text
formatting and is often utilized for formatting readme files and writing
messages in online discussion forums. Conversely, Gherkin is a plain text
language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as
a clue for editors, MDG files must have the `.feature.md` extension.

> </section>

Author: ```suggestion
```

> 
> </section>

Author: ```suggestion
```

<br /><br />

<section id="why-markdown-with-gherkin">

<!-- markdownlint-disable-next-line line-length -->
### Why Markdown with Gherkin&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#why-markdown-with-gherkin" id="ref-for-why-markdown-with-gherkin">⚓</a>

<br />

Markdown has become the de-facto standard for authoring rich text content in
plain text. A vast ecosystem of platforms and tools supports the editing and
rendering of Markdown documents.

> One can create beautiful documentation with Markdown, so it makes sense to

Author: ```suggestion
One can create stylish documentation with Markdown, so it makes sense to
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

> <section id="introduction">

Author: ```suggestion
<section id="intro">
```

### Introduction

<br />

The specification for the OpenINF Markdown dialect is based on several of its
more well-known predecessors, such as the [GitHub Flavored Markdown Spec][],
which is, in turn, based on the [CommonMark Spec][].

A <dfn id="spec">Specification</dfn> (<abbr>spec</abbr>) is a document that
outlines, in detail, how a technology or API is intended to function and how
it is accessed.

The OpenINF Markdown specification includes [many additions][] compared to the
GFM specification.

<br /><br />

> <section id="official-specifications">

Author: ```suggestion
<section id="official-specs">
```

> #### Official Specifications

Author: ```suggestion
#### Official Specifications&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#why-markdown-with-gherkin" id="ref-for-official-specs">⚓</a>
```

<br />

GFM and GLFM each have an official specification, which includes both:

1. The CommonMark standard.
1. Generic extensions to the CommonMark standard.

For example, GFM adds the [`strikethrough` extension][], and GLFM adds the
[`color chips` extension][]. These extensions in the official specifications are
not dependent upon any specific implementation or environment. They can be
implemented in any third-party Markdown rendering engine.

This document describes OpenINF Markdown with Gherkin (MDG), which some may
expect to detail how Gherkin support is added via generic extension(s) to the
CommonMark standard (as is typically the case). However, with MDG, inserting
Gherkin scenarios directly into Markdown using traditional Markdown syntax is
feasible. As a result, any GFM-compliant library can render MDG documents.

[`strikethrough` extension]: https://github.github.com/gfm/#strikethrough-extension-
[`color chips` extension]: https://docs.gitlab.com/ee/user/markdown.html#colors

> </section> <!-- END #official-specifications -->

Author: ```suggestion
</section> <!-- END #official-specs -->
```

<section id="termdef-mdg">
<!-- markdownlint-disable-next-line line-length -->

> <strong><em><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></em></strong>

Author: ```suggestion
<p><em><strong><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></strong></em>
```

is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of

> Gherkin</strong>.
> 
> </section> <!-- END #termdef-mdg -->

Author: ```suggestion
Gherkin</strong>.
</p></section> <!-- END #termdef-mdg -->
```

It is widely used for writing documentation, specifications, and test cases.
Markdown is a lightweight markup language that utilizes syntax for plain text
formatting and is often utilized for formatting readme files and writing
messages in online discussion forums. Conversely, Gherkin is a plain text
language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as
a clue for editors, MDG files must have the `.feature.md` extension.

> </section> <!-- END #introduction -->

Author: ```suggestion
</section> <!-- END #intro -->
```

<strong><em><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></em></strong>
is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.

</section> <!-- END #termdef-mdg -->

It is widely used for writing documentation, specifications, and test cases.
Markdown is a lightweight markup language that utilizes syntax for plain text
formatting and is often utilized for formatting readme files and writing
messages in online discussion forums. Conversely, Gherkin is a plain text
language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as
a clue for editors, MDG files must have the `.feature.md` extension.

> 
> </section> <!-- END #termdef-mdg -->

Author: ```suggestion
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

<section>

## OpenINF Markdown with Gherkin (MDG)

<br />  

> MDG, an acronym for Markdown with Gherkin, serves as a markup language that
> amalgamates the characteristic features of Markdown and Gherkin. It is a
> widely-used tool for writing documentation, specifications, and test cases.

Author: ```suggestion
<section id="termdef-mdg">
<!-- markdownlint-disable-next-line line-length -->
<strong><em><dfn data-dfn-type="termdef" id="def-for-termdef-mdg" role="button" aria-expanded="false" tabindex="0">Markdown with Gherkin (<abbr>MDG</abbr>) <a role="button" aria-expanded="false" tabindex="0" href="./#termdef-mdg" id="ref-for-termdef-mdg">⚓</a></dfn></em></strong>
is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.
</section>

It is widely used for writing documentation, specifications, and test cases.
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

> <section>

Author: ```suggestion
<section id="openinf-markdown-with-gherkin-mdg">
```

is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.
</section>

It is widely used for writing documentation, specifications, and test cases.
Markdown is a lightweight markup language that utilizes syntax for plain text
formatting and is often utilized for formatting readme files and writing
messages in online discussion forums. Conversely, Gherkin is a plain text
language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as
a clue for editors, MDG files must have the `.feature.md` extension. With MDG,
inserting Gherkin scenarios directly into Markdown using traditional Markdown
syntax is feasible. Moreover, any GFM-compliant library can render MDG
documents.

</section>

<br /><br />

> <section>

Author: ```suggestion
<section id="why-markdown-with-gherkin">
```

> ### Why Markdown with Gherkin

Author: ```suggestion
<!-- markdownlint-disable-next-line line-length -->
### Why Markdown with Gherkin&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#why-markdown-with-gherkin" id="why-markdown-with-gherkin-ref">⚓</a>
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

> <section id="openinf-markdown-with-gherkin-mdg">
> 
> ## OpenINF Markdown with Gherkin (MDG)
> 
> <br />  
> 
> <section id="termdef-mdg">

Author: ```suggestion
<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

<section id="introduction">

### Introduction

<br />

<section id="termdef-mdg">
```

is a markup language that amalgamates the <strong>expressive formatting
capabilities of Markdown</strong> with the <strong>conciseness and clarity of
Gherkin</strong>.
</section>

It is widely used for writing documentation, specifications, and test cases.
Markdown is a lightweight markup language that utilizes syntax for plain text
formatting and is often utilized for formatting readme files and writing
messages in online discussion forums. Conversely, Gherkin is a plain text
language deployed for writing acceptance tests, usually with the Behavior-Driven
Development (BDD) methodology. MDG is a Markdown version supported by the
Gherkin parser and serves as a strict superset of GitHub Flavored Markdown
(GFM)[^1]. To prevent Gherkin from processing regular `.md` files and serve as
a clue for editors, MDG files must have the `.feature.md` extension. With MDG,
inserting Gherkin scenarios directly into Markdown using traditional Markdown
syntax is feasible. Moreover, any GFM-compliant library can render MDG
documents.

</section>

<br /><br />

<section id="why-markdown-with-gherkin">

<!-- markdownlint-disable-next-line line-length -->

> ### Why Markdown with Gherkin&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#why-markdown-with-gherkin" id="why-markdown-with-gherkin-ref">⚓</a>

Author: ```suggestion
### Why Markdown with Gherkin&nbsp;<a role="button" aria-expanded="false" tabindex="0" href="#why-markdown-with-gherkin" id="ref-for-why-markdown-with-gherkin">⚓</a>
```

<!--
Previously, https://raw.githubusercontent.com/cucumber/gherkin/HEAD/MARKDOWN_WITH_GHERKIN.md
w/ credit going to original authors; however, adapted for strict adherence to the
OpenINF Community Markdown Style Guide Standards (minimal stylistic processing).
-->

<section id="openinf-markdown-with-gherkin-mdg">

## OpenINF Markdown with Gherkin (MDG)

<br />

**OpenINF Flavored Markdown Language Specification**

_1<sup>st</sup> edition (August 2023)_

---

<br /><br />

<section id="introduction">

### Introduction

<br />

> <section id="termdef-mdg">

Author: ```suggestion

The specification for the OpenINF Markdown dialect is based on several of
its more commonly-known predecessors, such as the [GitHub Flavored Markdown (GFM) specification][], which is in turn based on the [CommonMark specification][]. The OpenINF Markdown specification includes [many additions][] compared to the GFM specification.

<section id="termdef-mdg">
```

        | start | eat | left |
        |    12 |   5 |    7 |
        |    20 |   5 |   15 |

> ````

Author: ```suggestion
```
```

Markdown with Gherkin:

```markdown
## Feature: Staying Alive

This is about actually staying alive, not the
[Bee Gees song](https://www.youtube.com/watch?v=I_izvAbhExY).

### Rule: If You Don’t Eat, You Die

![xkcd](https://imgs.xkcd.com/comics/lunch_2x.png)

`@important` `@essential`
#### Scenario Outline: Eating

- Given there are <start> cucumbers
- When I eat <eat> cucumbers
- Then I should have <left> cucumbers

##### Examples:

| start      | eat | left      |
| ---------- | --- | --------- |
| 12         | 5   | 7         |
| 20         | 5   | 15        |
```

This Markdown document can be rendered beautifully by any GFM-compliant tool.

</section>

<br /><br />

<section>

### The Parsing Rules

<br />

The following keywords (and their translations) must be preceded by one or more

> hash marks (`#`) (Markdown header). For example:

Author: ```suggestion
hash marks (`#`) (Markdown heading). For example:
```

- Feature

>   - The Feature header is optional in MDG. If the document does not start with

Author: ```suggestion
  - The **Feature** heading is optional in MDG. If the document does not start with
```

The following step keywords (and their translations) must be preceded by a
hyphen (`-`) or an asterisk (`*`) (Markdown list item). For example:

- `- Given`
- `- When`
- `- Then`
- `- And`
- `- But`

Important: Data Tables and Examples Tables must be indented 2-5 spaces to be
recognized. Unindented GFM tables will not be recognized as Data Tables or
Examples Tables.

- Data Tables and Examples tables use the [GFM table][] syntax.
- Doc Strings use the [GFM fenced code blocks][] syntax.
- Tags must be wrapped by a single backtick (`` ` ``) on each side, for
  example, `` `@hello` ``, and be placed on a line above the keyword (as with
  Gherkin Classic).

</section>

<br /><br />

<section>

### Rendering MDG With Results

<br />

The [@cucumber/react][] library provides a `<MDG>` React component that renders
a Markdown document with results from Cucumber (using [Cucumber Messages][]).

The rendered HTML contains the original contents from the Markdown file, with
steps and scenarios colored according to the results from the message stream. It
also renders attachments from Cucumber (screenshots, videos, logs, etc.).

</section>

<br /><br />

> <section>

Author: ```suggestion
<section id="some-notes-about-parsing-mdg">
```

### Some Notes About Parsing MDG

<br />

Both Gherkin Classic and MDG documents utilize the `Parser` class for parsing,
but they have different TokenMatchers (scanners/lexers). Markdown documents
often contain lines that are not relevant to Gherkin, such as prose, which is
why the `GherkinInMarkdownTokenMatcher` treats all unrecognized lines as a
special token labeled as _Empty_. As a result, the `GherkinDocument` AST will
have empty description properties, which means the scenario description property
will not be included in the JSON formatter.

</section>

<br /><br />

> <section>

Author: ```suggestion
<section id="acknowledgements">
```

will not be included in the JSON formatter.

</section>

<br /><br />

<section id="acknowledgements">

### Acknowledgements

<br />

We express our gratitude to the authors and contributors of the [@cucumber][]
project. This work is based on the original [Cucumber Markdown with Gherkin][]
but has been modified to comply with the OpenINF Community Markdown Style Guide
Standards strictly. The changes include minimal stylistic processing and
significant text restructuring, such as correcting spelling errors and document
organization.

</section>

<br /><br />

<!-- BEGIN LINK DEFINITIONS -->

[^1]:
    Support for MDG is currently only supported in the [@cucumber/gherkin][]
    (the JavaScript implementation), version `19.0.0` and above. The Cucumber
    team will gather feedback from users of [@cucumber/cucumber][] before
    porting MDG to other programming languages.

[@cucumber]: https://github.com/cucumber
[@cucumber/cucumber]: https://www.npmjs.com/package/@cucumber/cucumber
[@cucumber/gherkin]: https://www.npmjs.com/package/@cucumber/gherkin
[@cucumber/react]:
  https://github.com/cucumber/cucumber-react
  "React components for Cucumber"

> [Cucumber Markdown with Gherkin]:

Author: ```suggestion
[CommonMark Spec]: https://spec.commonmark.org
[Cucumber Markdown with Gherkin]:
```

but has been modified to comply with the OpenINF Community Markdown Style Guide
Standards strictly. The changes include minimal stylistic processing and
significant text restructuring, such as correcting spelling errors and document
organization.

</section>

<br /><br />

<!-- BEGIN LINK DEFINITIONS -->

[^1]:
    Support for MDG is currently only supported in the [@cucumber/gherkin][]
    (the JavaScript implementation), version `19.0.0` and above. The Cucumber
    team will gather feedback from users of [@cucumber/cucumber][] before
    porting MDG to other programming languages.

[@cucumber]: https://github.com/cucumber
[@cucumber/cucumber]: https://www.npmjs.com/package/@cucumber/cucumber
[@cucumber/gherkin]: https://www.npmjs.com/package/@cucumber/gherkin
[@cucumber/react]:
  https://github.com/cucumber/cucumber-react
  "React components for Cucumber"
[CommonMark Spec]: https://spec.commonmark.org
[Cucumber Markdown with Gherkin]:
  https://github.com/cucumber/gherkin/blob/main/MARKDOWN_WITH_GHERKIN.md?plain=1
  "Cucumber Markdown with Gherkin"
[Cucumber Messages]:
  https://github.com/cucumber/messages/blob/main/messages.md
  "Cucumber Messages"

> [GFM table]:

Author: ```suggestion
[GitHub Flavored Markdown Spec]:
  https://github.github.com/gfm/
  "GitHub Flavored Markdown (GFM) formal specification"
[GFM table]:
```

will not be included in the JSON formatter.

</section>

<br /><br />

<section id="acknowledgements">

### Acknowledgements

<br />

We express our gratitude to the authors and contributors of the [@cucumber][]
project. This work is based on the original [Cucumber Markdown with Gherkin][]
but has been modified to comply with the OpenINF Community Markdown Style Guide
Standards strictly. The changes include minimal stylistic processing and
significant text restructuring, such as correcting spelling errors and document
organization.

</section>

<br /><br />

<!-- BEGIN LINK DEFINITIONS -->

[^1]:
    Support for MDG is currently only supported in the [@cucumber/gherkin][]
    (the JavaScript implementation), version `19.0.0` and above. The Cucumber
    team will gather feedback from users of [@cucumber/cucumber][] before
    porting MDG to other programming languages.

[@cucumber]: https://github.com/cucumber
[@cucumber/cucumber]: https://www.npmjs.com/package/@cucumber/cucumber
[@cucumber/gherkin]: https://www.npmjs.com/package/@cucumber/gherkin
[@cucumber/react]:
  https://github.com/cucumber/cucumber-react
  "React components for Cucumber"

> [CommonMark Spec]: https://spec.commonmark.org

Author: ```suggestion
[CommonMark Spec]:
  https://spec.commonmark.org
  "CommonMark formal specification"
```

<br />

The [@cucumber/react][] library provides a `<MDG>` React component that renders
a Markdown document with results from Cucumber (using [Cucumber Messages][]).

The rendered HTML contains the original contents from the Markdown file, with
steps and scenarios colored according to the results from the message stream. It
also renders attachments from Cucumber (screenshots, videos, logs, etc.).

</section>

<br /><br />

<section id="some-notes-about-parsing-mdg">

### Some Notes About Parsing MDG

<br />

Both Gherkin Classic and MDG documents utilize the `Parser` class for parsing,
but they have different TokenMatchers (scanners/lexers). Markdown documents
often contain lines that are not relevant to Gherkin, such as prose, which is
why the `GherkinInMarkdownTokenMatcher` treats all unrecognized lines as a
special token labeled as _Empty_. As a result, the `GherkinDocument` AST will
have empty description properties, which means the scenario description property
will not be included in the JSON formatter.

</section>

<br /><br />

<section id="acks">

### Acknowledgements

<br />

We express our gratitude to the authors and contributors of the [@cucumber][]

> project. This work is based on the original [Cucumber Markdown with Gherkin][]

Author: ```suggestion
project. This work is based on the original [Cucumber Markdown with Gherkin][],
```

> but has been modified to comply with the OpenINF Community Markdown Style Guide

Author: ```suggestion
but has been modified to comply with the [OpenINF Community Markdown Style Guide][].
```

> Standards strictly. The changes include, amongst others, minimal stylistic processing,

Author: ```suggestion
The changes include, amongst others, minimal stylistic processing,
```

> We express our gratitude to the authors and contributors of the [@cucumber][]
> project. This work is based on the original [Cucumber Markdown with Gherkin][],
> but has been modified to comply with the [OpenINF Community Markdown Style Guide][].
> The changes include, amongst others, minimal stylistic processing,
> significant text restructurings, document re-organization, and inclusion of foreign
> content ethically sourced from their properly-attributed authoritative sources.

Author: ```suggestion
We express our gratitude to the authors and contributors of the [@cucumber][] project. This work is
based on the original [Cucumber Markdown with Gherkin][], but has been modified to comply with the
[OpenINF Community Markdown Style Guide][]. The changes include, amongst others, minimal stylistic
processing, significant text restructurings, document re-organization, and inclusion of foreign
content ethically sourced from their properly-attributed authoritative sources.
```

Markdown with Gherkin:

```markdown
## Feature: Staying Alive

This is about actually staying alive, not the
[Bee Gees song](https://www.youtube.com/watch?v=I_izvAbhExY).

### Rule: If You Don’t Eat, You Die

![xkcd](https://imgs.xkcd.com/comics/lunch_2x.png)

`@important` `@essential`
#### Scenario Outline: Eating

- Given there are <start> cucumbers
- When I eat <eat> cucumbers
- Then I should have <left> cucumbers

##### Examples:

| start      | eat | left      |
| ---------- | --- | --------- |
| 12         | 5   | 7         |
| 20         | 5   | 15        |
```

This Markdown document can be rendered on-the-fly by any GFM-compliant tool.

</section> <!-- END #why-markdown-with-gherkin -->

<br /><br />

> <section>

Author: ```suggestion
<section id="parsing-rules">
```

- Rule: The rule of this document is that all keywords must be preceded by one
  or more hash marks (`#`).
- Scenario: The scenario of this document is that a user wants to create a
  document using MDG.
- Scenario Outline: The scenario outline of this document is that a user wants
  to create a document that includes several sections.
- Examples: The examples of this document are:
  - A document that describes the use of keywords in MDG.
  - A document that includes several different sections.

The following step keywords (and their translations) must be preceded by a
hyphen (`-`) or an asterisk (`*`) (Markdown list item). For example:

- `- Given`
- `- When`
- `- Then`
- `- And`
- `- But`

Important: Data Tables and Examples Tables must be indented 2-5 spaces to be
recognized. Unindented GFM tables will not be recognized as Data Tables or
Examples Tables.

- Data Tables and Examples tables use the [GFM table][] syntax.
- Doc Strings use the [GFM fenced code blocks][] syntax.
- Tags must be wrapped by a single backtick (`` ` ``) on each side, for
  example, `` `@hello` ``, and be placed on a line above the keyword (as with
  Gherkin Classic).

</section>

<br /><br />

> <section>

Author: ```suggestion
<section id="rendering-mdg-with-results">
```

### Rendering MDG With Results

<br />

The [@cucumber/react][] library provides a `<MDG>` React component that renders
a Markdown document with results from Cucumber (using [Cucumber Messages][]).

The rendered HTML contains the original contents from the Markdown file, with
steps and scenarios colored according to the results from the message stream. It
also renders attachments from Cucumber (screenshots, videos, logs, etc.).

</section>

<br /><br />

<section id="some-notes-about-parsing-mdg">

### Some Notes About Parsing MDG

<br />

Both Gherkin Classic and MDG documents utilize the `Parser` class for parsing,
but they have different TokenMatchers (scanners/lexers). Markdown documents
often contain lines that are not relevant to Gherkin, such as prose, which is
why the `GherkinInMarkdownTokenMatcher` treats all unrecognized lines as a
special token labeled as _Empty_. As a result, the `GherkinDocument` AST will

> have empty description properties, which means the scenario description property

Author: ```suggestion
have _empty description properties_, which means the scenario description property
```

> have _empty description properties_, which means the scenario description property

Author: ```suggestion
have _empty description properties_, which means the scenario `description` property
```

will not be included in the JSON formatter.

> 

Author: ```suggestion

<!-- TODO(DerekNonGeneric): link to JSON5 formatter -->

```

> </section>

Author: ```suggestion

For this reason, the `GherkinDocument` AST will have _empty description
properties_. This means that the JSON formatter will not include a `description`
property for scenarios.

</section>
```

<br /><br />

<section id="acks">

### Acknowledgements

<br />

We express our gratitude to the authors and contributors of the [@cucumber][] project. This work is
based on the original [Cucumber Markdown with Gherkin][], but has been modified to comply with the
[OpenINF Community Markdown Style Guide][]. The changes include, amongst others, minimal stylistic
processing, significant text restructurings, document re-organization, and inclusion of foreign
content ethically sourced from their properly-attributed authoritative sources.

</section>

<br /><br />

> <!-- BEGIN LINK DEFINITIONS -->

Author: ```suggestion
<!-- LINK LABEL DEFINITIONS - START -->
```

    (the JavaScript implementation), version `19.0.0` and above. The Cucumber
    team will gather feedback from users of [@cucumber/cucumber][] before
    porting MDG to other programming languages.

[@cucumber]: https://github.com/cucumber
[@cucumber/cucumber]: https://www.npmjs.com/package/@cucumber/cucumber
[@cucumber/gherkin]: https://www.npmjs.com/package/@cucumber/gherkin
[@cucumber/react]:
  https://github.com/cucumber/cucumber-react
  "React components for Cucumber"
[CommonMark Spec]:
  https://spec.commonmark.org
  "CommonMark formal specification"
[Cucumber Markdown with Gherkin]:
  https://github.com/cucumber/gherkin/blob/main/MARKDOWN_WITH_GHERKIN.md?plain=1
  "Cucumber Markdown with Gherkin"
[Cucumber Messages]:
  https://github.com/cucumber/messages/blob/main/messages.md
  "Cucumber Messages"
[GitHub Flavored Markdown Spec]:
  https://github.github.com/gfm/
  "GitHub Flavored Markdown (GFM) formal specification"
[GFM table]:
  https://github.github.com/gfm/#tables-extension-  
[GFM fenced code blocks]:
  https://github.github.com/gfm/#fenced-code-blocks
[living documentation]: https://leanpub.com/bddbooks-formulation

> <!-- END LINK DEFINITIONS -->

Author: ```suggestion
<!-- LINK LABEL DEFINITIONS - END -->
```

Markdown with Gherkin:

```markdown
## Feature: Staying Alive

This is about actually staying alive, not the
[Bee Gees song](https://www.youtube.com/watch?v=I_izvAbhExY).

### Rule: If You Don’t Eat, You Die

![xkcd](https://imgs.xkcd.com/comics/lunch_2x.png)

`@important` `@essential`
#### Scenario Outline: Eating

- Given there are <start> cucumbers
- When I eat <eat> cucumbers
- Then I should have <left> cucumbers

##### Examples:

| start      | eat | left      |
| ---------- | --- | --------- |
| 12         | 5   | 7         |
| 20         | 5   | 15        |
```

> This Markdown document can be rendered beautifully by any GFM-compliant tool.

Author: ```suggestion
This Markdown document can be rendered on-the-fly by any GFM-compliant tool.
```

> </section>

Author: ```suggestion
</section> <!-- END #why-markdown-with-gherkin -->
```

  Gherkin Classic).

</section>

<br /><br />

<section>

### Rendering MDG With Results

<br />

The [@cucumber/react][] library provides a `<MDG>` React component that renders
a Markdown document with results from Cucumber (using [Cucumber Messages][]).

The rendered HTML contains the original contents from the Markdown file, with
steps and scenarios colored according to the results from the message stream. It
also renders attachments from Cucumber (screenshots, videos, logs, etc.).

</section>

<br /><br />

<section id="some-notes-about-parsing-mdg">

### Some Notes About Parsing MDG

<br />

Both Gherkin Classic and MDG documents utilize the `Parser` class for parsing,
but they have different TokenMatchers (scanners/lexers). Markdown documents
often contain lines that are not relevant to Gherkin, such as prose, which is
why the `GherkinInMarkdownTokenMatcher` treats all unrecognized lines as a
special token labeled as _Empty_. As a result, the `GherkinDocument` AST will
have empty description properties, which means the scenario description property
will not be included in the JSON formatter.

</section>

<br /><br />

> <section id="acknowledgements">

Author: ```suggestion
<section id="acks">
```

### Acknowledgements

<br />

We express our gratitude to the authors and contributors of the [@cucumber][]
project. This work is based on the original [Cucumber Markdown with Gherkin][]
but has been modified to comply with the OpenINF Community Markdown Style Guide

> Standards strictly. The changes include minimal stylistic processing and

Author: ```suggestion
Standards strictly. The changes include, amongst other, stylistic processing and
```

> Standards strictly. The changes include, amongst other, stylistic processing and
> significant text restructuring, such as correcting spelling errors and document
> organization.

Author: ```suggestion
Standards strictly. The changes include, amongst others, minimal stylistic processing,
significant text restructurings, document re-organization, and inclusion of foreign
content sourced from authoritative sources.
```

  example, `` `@hello` ``, and be placed on a line above the keyword (as with
  Gherkin Classic).

</section>

<br /><br />

<section>

### Rendering MDG With Results

<br />

The [@cucumber/react][] library provides a `<MDG>` React component that renders
a Markdown document with results from Cucumber (using [Cucumber Messages][]).

The rendered HTML contains the original contents from the Markdown file, with
steps and scenarios colored according to the results from the message stream. It
also renders attachments from Cucumber (screenshots, videos, logs, etc.).

</section>

<br /><br />

<section id="some-notes-about-parsing-mdg">

### Some Notes About Parsing MDG

<br />

Both Gherkin Classic and MDG documents utilize the `Parser` class for parsing,
but they have different TokenMatchers (scanners/lexers). Markdown documents
often contain lines that are not relevant to Gherkin, such as prose, which is
why the `GherkinInMarkdownTokenMatcher` treats all unrecognized lines as a
special token labeled as _Empty_. As a result, the `GherkinDocument` AST will
have empty description properties, which means the scenario description property
will not be included in the JSON formatter.

> For this reason, the `GherkinDocument` AST will have _empty description
> properties_. This means that the JSON formatter will not include a `description`
> property for scenarios.
> 

Author: ```suggestion
```
